config/v1/types_infrastructure.go

```diff
@@ -1082,9 +1082,42 @@ type VSpherePlatformStatus struct {
 	LoadBalancer *VSpherePlatformLoadBalancer `json:"loadBalancer,omitempty"`
 }
 
+// IBMCloudServiceEndpoint stores the configuration of a custom url to
+// override existing defaults of IBM Cloud Services.
+type IBMCloudServiceEndpoint struct {
+	// name is the name of the IBM Cloud service.
+	// For example, the IBM Cloud Private IAM service could be configured with the
+	// service `name` of `Iam` and `url` of `https://private.iam.cloud.ibm.com`
+	// Whereas the IBM Cloud Private VPC service for US South (Dallas) could be configured
+	// with the service `name` of `Vpc` and `url` of `https://us.south.private.iaas.cloud.ibm.com`
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
+	// +kubebuilder:validation:MaxLength=32
+	Name string `json:"name"`
+
+	// url is fully qualified URI with scheme https, that overrides the default generated
+	// endpoint for a client.
+	// This must be provided and cannot be empty.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=uri
+	// +kubebuilder:validation:Pattern=`^https://`
```

Comment:
You can now use [CEL validation](https://kubernetes.io/docs/reference/using-api/cel/#kubernetes-url-library) for URLs, lets swap to that as it will be a more complete validation

Reply:
I was digging around the docs for an example, without success. Is this simply done via,
`+kubebuilder:validation:Format=url`?
I started with this, hoping it was the correct expectation.

Reply:
I expect something like below to work
```
// +kubebuilder:validation:XValidation:rule="isURL(self)",message="url must be a valid absolute URL"
```

Reply:
updated